packages/flutter_tools/lib/src/asset.dart

```diff
@@ -203,6 +231,25 @@ class _Asset {
 
   @override
   String toString() => 'asset: $assetEntry';
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(other, this))
+      return true;
+    if (other is! _Asset)
+      return false;
+    return base == other.base &&
```

Comment:
nit: vertically align the expressions (you can put the && at the start of the line too to make it prettier)